src/main/java/bio/terra/service/SamClientService.java

```diff
@@ -189,6 +190,11 @@ public String createDatasetResource(AuthenticatedUserRequest userReq, UUID datas
         req.addPoliciesItem(
             DataRepoRole.READER.getPolicyName(),
             new AccessPolicyMembership().roles(Collections.singletonList(DataRepoRole.READER.getRoleName())));
+        if (!readersList.isPresent()) {
+            req.addPoliciesItem(
+                DataRepoRole.READER.getPolicyName(),
+                new AccessPolicyMembership().roles(readersList.get()));
```

Comment:
We want to be setting member emails here using the `readersList`, not the roles. I believe we'll need to use the sam client to either [overwrite the memberEmail](https://sam.dsde-dev.broadinstitute.org/#!/Resources/overwritePolicyMembers) or go through the list and [add one member email at a time](https://sam.dsde-dev.broadinstitute.org/#!/Resources/addUserToPolicy). Since this is when the dataset is first created I prefer the first option: one call and no risk of overwriting a reader that already exists.